hi-ml-azure/src/health/azure/azure_util.py

```diff
@@ -716,56 +718,69 @@ def get_run_file_names(run: Run, prefix: str = "") -> List[str]:
     return [f for f in all_files if f.startswith(prefix)] if prefix else all_files
 
 
-def download_run_files(run: Run, output_dir: Path, prefix: str = "") -> None:
+def _download_files_from_run(run: Run, output_dir: Path, prefix: str = "", validate_checksum: bool = False) -> None:
     """
-    Download all files for a given run, which optionally start with a given prefix
+    Download all files for a given AML run, where the filenames may optionally start with a given
+    prefix.
 
     :param run: The AML Run to download associated files for
     :param output_dir: Local directory to which the Run files should be downloaded.
     :param prefix: Optional prefix to filter Run files by
+    :param validate_checksum: Whether to validate the content from HTTP response
     """
     run_paths = get_run_file_names(run, prefix=prefix)
     if len(run_paths) == 0:
         raise ValueError("No such files were found for this Run.")
 
     for run_path in run_paths:
         output_path = output_dir / run_path
-        download_run_file(run, run_path, output_path)
+        _download_file_from_run(run, run_path, output_path, validate_checksum=validate_checksum)
 
 
-def download_run_files_from_run_id(run_id: str, output_dir: Path, prefix: str = "",
-                                   workspace: Optional[Workspace] = None,
-                                   workspace_config_path: Optional[Path] = None) -> None:
+def download_files_from_run_id(run_id: str, output_folder: Path, prefix: str = "",
+                               workspace: Optional[Workspace] = None,
+                               workspace_config_path: Optional[Path] = None,
+                               validate_checksum: bool = False) -> None:
     """
-    For a given Azure ML run id, first retrieve the Run, and then download all files,
-    which optionally start with a given prefix
+    For a given Azure ML run id, first retrieve the Run, and then download all files, which optionally start
+    with a given prefix. E.g. if the Run creates a folder called "outputs", which you wish to download all
+    files from, specify prefix="outputs". To download all files associated with the run, leave prefix as empty.
```

Comment:
Maybe add documentation here and in download_checkpoints to describe how the local file names are created?
If I have a run with files "outputs/foo/bar/file.txt", and I specify prefix="outputs/foo", what will my local file be called? will I get output_folder/outputs/foo/bar/file.txt, or something else?